internal/exec/stages/mount/mount.go

```diff
@@ -117,10 +118,19 @@ func (s stage) mountFs(fs types.Filesystem) error {
 		return err
 	}
 
-	if err := s.Logger.LogOp(func() error { return syscall.Mount(fs.Device, path, *fs.Format, 0, "") },
-		"mounting %q at %q with type %q", fs.Device, path, *fs.Format,
+	args := translateOptionSliceToString(fs.MountOptions, ",")
+	if err := s.Logger.LogOp(func() error { return syscall.Mount(fs.Device, path, *fs.Format, 0, args) },
```

Comment:
Sorry for the late review; I was at a conference then some regressions prevented me from building an image to test for a bit. Looks like this only works with FS specific options, but not ones that end up as flags for the fourth argument. For example, `ikeep` is xfs specific and works, but `noexec` is not and doesn't since that ends up as the flag `MS_NOEXEC`. So this needs to learn how to handle those separately. It would be good to add a blackbox test for both types of args as well.

Reply:
https://github.com/karelzak/util-linux/blob/master/libmount/src/optstr.c#L651 and that file in general looks useful, though it would need cgo.

Reply:
Just that it's not part of Go, but an additional dependency. I wasn't aware that `syscall` is deprecated, so this is indeed a non-issue.

Reply:
It'd be good to not vendor something not really meant as a library especially something large like docker, so I think I'd prefer exec'ing mount if you're opposed to the cgo implementation. @lucab do you have strong opinions here?

Reply:
I'm not opposed to use _libmount_, it's just that I usually try to find a native library before implementing something myself.

FYI: The size of the Ignition executable is increasing by ~100 KB when using the Docker module.

Reply:
It's less about the actual binary size and more about needing to keep up to date with another dependency as well as the go module tooling being bad and wanting to pull in more than necessary to `vendor/`.

Reply:
@ajeddeloh either directions are fine. As we are already shelling out to `mount` in other places, I think that would be path of least resistance here too.

Reply:
Implemented as a shell command as suggested.